### ISSUE TITLE
CSV parser assumes (unquoted) IP addresses are floating point

### ISSUE THREAD

**Issue by rheatley-pervasid** [zpl-c/zpl#98]
Consider the following
```
zpl_csv_object root = { 0 };
char ip_test[255] = "123.45.67.89";
zpl_u8 err = zpl_csv_parse(&root, ip_test, zpl_heap_allocator(), false);
if (err)
     printf("zpl_csv_parse() failed %u", err);
else
    zpl_json_write(zpl_file_get_standard(ZPL_FILE_STANDARD_OUTPUT), &root, 0);
```

The output is as follows
```
[
    [123.450000]
]
```

My understanding is the test in `zpl_csv_parse_delimiter` is too naive.

For now I have added a change to `zpl_adt_parse_number` (after the "false positives and special cases")
```
/* skip ip addresses */
zpl_u8 dots = 0;
char *dots_p = p;
do {
    if (*dots_p == '.' && ++dots > 1)
        return ++base_str;
} while (*dots_p++);
```
But this is also a bit naive if you are worried about performance!

**Comment by zpl-zak** [comment]
Hi, thanks for the report; a patch will be made to detect and handle such occasions appropriately.

**Comment by rheatley-pervasid** [comment]
Great - so far my experience with ZPL is extremely positive and it seems awesome. So thank you very much!

**Comment by zpl-zak** [comment]
This issue is now resolved in https://github.com/zpl-c/zpl/commit/0d3118bac63f188c1e1763952076a3bfa1f4f61a and released into zpl version [18.0.1](https://github.com/zpl-c/zpl/releases/tag/18.0.1)
